src/test/java/org/folio/settings/server/main/MainVerticleTest.java

```diff
@@ -727,6 +727,38 @@ public void testGetSettings() {
         .contentType(ContentType.TEXT);
   }
 
+  @Test
+  public void testGetSettingsStream() {
+    JsonObject en = new JsonObject()
+        .put("scope", UUID.randomUUID().toString())
+        .put("value", new JsonObject().put("v", "stream"));
+    JsonArray permGlobalWrite = new JsonArray().add("mod-settings.global.write." + en.getString("scope"));
+    JsonArray permGlobalRead = new JsonArray().add("mod-settings.global.read." + en.getString("scope"));
+    for (int i = 0; i < 201; i++) {
```

Comment:
Maybe add a comment for why you're doing this large number of iterations, perhaps linking to the ticket.

Reply:
If you run this test against the old code with the presumed problem, does this replicate the problem?

Reply:
From the [comments on the ticket](https://folio-org.atlassian.net/browse/MODSET-17?focusedCommentId=233126) it looks like this code is being tested in the environment where the error is occurring and it hasn't fixed it, which makes me think as Julian has implied it may be something with this environment's postgres setup.

I don't have anything against merging this code as it seems to have some perfectly good hardening however but maybe we should keep open if other ideas occur to us about potential code fixes.